src/components/TonConnectUIProvider.vue

```diff
@@ -39,7 +40,7 @@ export default defineComponent({
     );
 
     // Provide the TonConnectUI instance to child components
-    provide("tonConnectUI", tonConnectUI.value);
+    provide(tonConnectUIKey, tonConnectUI.value);
```

Comment:
Actually, as I see above, `tonConnectUI.value.uiOptions` are updated using watch. So instance of TonConnectUI is the same. Therefore, you don't need to subscribe for `tonConnectUI.value` updates in consumers (eg. useTonWallet).

So just remove useless code from that component (`wallet`, `unsubscribe`, etc.)